modules/digitaluniverse/rendering/renderablebillboardscloud.cpp

```diff
@@ -1531,17 +1494,24 @@ void RenderableBillboardsCloud::createDataSlice() {
         minColorIdx = colorIdx < minColorIdx ? colorIdx : minColorIdx;
     }
 
+    double maxRadius = 0.0;
+
     float biggestCoord = -1.f;
     for (size_t i = 0; i < _fullData.size(); i += _nValuesPerAstronomicalObject) {
-        glm::dvec4 transformedPos = _transformationMatrix * glm::dvec4(
+        glm::vec3 transformedPos = glm::vec3(_transformationMatrix * glm::vec4(
             _fullData[i + 0],
             _fullData[i + 1],
             _fullData[i + 2],
             1.0
-        );
-        // W-normalization
-        transformedPos /= transformedPos.w;
-        glm::vec4 position(glm::vec3(transformedPos), static_cast<float>(_unit));
+        ));
+        glm::vec4 position(transformedPos, static_cast<float>(_unit));
+
+        const double unitMeter = unitToMeter(_unit);
+        glm::dvec3 p = glm::dvec3(glm::dvec3(position) * unitMeter);
```

Comment:
You could also just do:
`glm::dvec3 p = glm::dvec3(position * unitMeter);`

Reply:
Had to do `glm::dvec3(position) * unitMeter` as `glm::dvec3(position * unitMeter)` doesn't compile since position is glm::vec3 and glm::vec3 * double doesn't work